Build failure: 'MM_MemorySubSpaceTarok::reset' hides overloaded virtual function

OpenJ9 fails to build with https://github.com/eclipse-openj9/openj9-omr/commit/7f732bfe8d9b113c07e0faf0aac482dc65c4e9f1. Build error is:

```
07:59:19  In file included from /Users/jenkins/workspace/Build_JDK17_aarch64_mac_OMR/openj9/runtime/gc_vlhgc/AllocationContextBalanced.cpp:41:
07:59:19  /Users/jenkins/workspace/Build_JDK17_aarch64_mac_OMR/openj9/runtime/gc_vlhgc/MemorySubSpaceTarok.hpp:250:15: error: 'MM_MemorySubSpaceTarok::reset' hides overloaded virtual function [-Werror,-Woverloaded-virtual]
07:59:19          virtual void reset();
07:59:19                       ^
07:59:19  /Users/jenkins/workspace/Build_JDK17_aarch64_mac_OMR/omr/gc/base/MemorySubSpace.hpp:346:15: note: hidden overloaded virtual function 'MM_MemorySubSpace::reset' declared here: different number of parameters (1 vs 0)
07:59:19          virtual void reset(MM_EnvironmentBase *env = NULL);
07:59:19                       ^
```

Fails on aarch64 Linux JDK8 (but not 11, 17, 21), aarch64 Mac JDK 11, 17, 21, x86-64 Mac JDK 8, 11, 17, 21. 

I suspect the problem is with https://github.com/eclipse-omr/omr/pull/8284, FYI @amicic.

--- a/runtime/gc_vlhgc/MemorySubSpaceTarok.cpp
+++ b/runtime/gc_vlhgc/MemorySubSpaceTarok.cpp
@@ -428,6 +428,14 @@ MM_MemorySubSpaceTarok::reset()
 	Assert_MM_unreachable();
 }
 
+void
+MM_MemorySubSpaceTarok::reset(MM_EnvironmentBase *env)
+{
+	/* unused in Tarok collectors */
+	Assert_MM_unreachable();
+}
+
+
 /**
  * As opposed to reset, which will empty out, this will fill out as if everything is free
  */

--- a/runtime/gc_vlhgc/MemorySubSpaceTarok.hpp
+++ b/runtime/gc_vlhgc/MemorySubSpaceTarok.hpp
@@ -248,6 +248,7 @@ public:
 	virtual MM_MemorySubSpace *getTenureMemorySubSpace();
 
 	virtual void reset();
+	virtual void reset(MM_EnvironmentBase *env);
 	virtual void rebuildFreeList(MM_EnvironmentBase *env);
 	
 	virtual void resetLargestFreeEntry();

